dao/src/main/java/org/thingsboard/server/dao/sql/attributes/JpaAttributeDao.java

```diff
@@ -78,7 +82,9 @@ private void init() {
                 .maxDelay(maxDelay)
                 .statsPrintIntervalMs(statsPrintIntervalMs)
                 .build();
-        queue = new TbSqlBlockingQueue<>(params);
+
+        Function<AttributeKvEntity, Integer> hashcodeFunction = entity -> entity != null ? entity.getId().getEntityId().hashCode() : 0;
```

Comment:
put null check to the     public ListenableFuture<Void> add(E element)  method to avoid code duplication.